### ISSUE TITLE
DiscordCallback typings incorrect

### ISSUE THREAD

**Issue by roncli** [brussell98/discord-markdown#29]
Here is the interface for DiscordCallback

```
interface DiscordCallback {
  user?: (id: number) => string;
  channel?: (id: number) => string;
  role?: (id: number) => string;
  everyone?: () => string;
  here?: () => string;
}
```

However, this is incorrect.  None of the first three actually take an id as a parameter.  `user` takes a user object, `channel` takes a channel object, and `role` takes a role object.  See:

```
const discordCallbackDefaults = {
	user: node => '@' + markdown.sanitizeText(node.id),
	channel: node => '#' + markdown.sanitizeText(node.id),
	role: node => '&' + markdown.sanitizeText(node.id),
	everyone: () => '@everyone',
	here: () => '@here'
};
```

Could these typings be fixed to reflect the actual usage for the callback?

**Comment by brussell98** [comment]
~~This is incorrect. The callbacks are called with a node containing only the id parsed from the mention.~~
The incorrect typings are causing by me not releasing a new version with the fix. v2.5.1 has the correct types.

**Comment by roncli** [comment]
Ah, I see what you're missing now.  Take a closer look at the TypeScript definition of `user`, `channel`, and `role`:

`(id: number) => string`

It looks like you are thinking that this means the parameter is an object with a key of `id`.  However, without curly braces `{ }`, this is not what is happening.  TypeScript thinks you're defining a function with one parameter named `id` that is a number.

Here is what it is thrown in Visual Studio Code, demonstrating the problem.  Note that this code runs correctly in node.js, it's just TypeScript throwing the error.

![image](https://user-images.githubusercontent.com/498929/133828987-7acacf44-bbd5-4106-8bfb-c6bce196b940.png)

If you want to have that first parameter be an object, you need curly braces:

`({id: number}) => string`

You can even name that parameter `node` to be consistent with the defaults:

`(node: {id: number}) => string`

This is still incorrect, however, because `id` numbers coming from Discord are _strings_, because they are too long to represent as numbers.  As an example of this, here is the ID number of my server, Overload Teams League, and what node does if it assumes it's a number:

![image](https://user-images.githubusercontent.com/498929/133828812-2c577bc9-0ea2-47b1-8d11-86448f55643b.png)

So, you will want to change the type of `id` to a string.

`(node: {id: string}) => string`

I hope this clears things up!

**Comment by roncli** [comment]
OH! Never mind!  You fixed this in 2.5.1! :)
